packages/demo-protobuf/src/registry.ts

```diff
@@ -0,0 +1,30 @@
+import protobuf from "protobufjs";
+
+import { cosmos_sdk as cosmosSdk, google } from "./generated/codecimpl";
+
+export type GeneratedType = {
+  readonly create: (properties?: { [k: string]: any }) => protobuf.Message<{}>;
+  readonly encode: (
+    message: protobuf.Message<{}> | { [k: string]: any },
+    writer?: protobuf.Writer,
+  ) => protobuf.Writer;
+  readonly decode: (reader: protobuf.Reader | Uint8Array, length?: number) => protobuf.Message<{}>;
+};
+
+export class Registry {
+  private readonly types: Map<string, GeneratedType>;
+
+  constructor(customTypes: Iterable<[string, GeneratedType]> = []) {
+    this.types = new Map<string, GeneratedType>([
+      ["/cosmos.Coin", (cosmosSdk.v1.Coin as unknown) as GeneratedType],
```

Comment:
Nice! This should work without casting now, right?

Reply:
It's not so easy. See https://github.com/CosmWasm/cosmjs/pull/221/commits/f45cc06a5c0781c72d36e509484cca02a1d482ff for a first attempt.